### PR TITLE
Make options.bottommost_compression, compression_opts and bottommost_compression_opts dynamically changeable.

### PR DESCRIPTION
**Pull request by siying**
Summary:
These three options should be made dynamically changeable. Simply add them to MutableCFOptions and made the change.

Test Plan: Add a unit test to make sure that SetOptions() can change the options.

### CODE REVIEW

File: options/options_helper.cc [hunk 1]
```diff
@@ -786,37 +786,117 @@ bool SerializeSingleOptionHelper(const char* opt_address,
...
+    if (start >= value.size()) {
+      return Status::InvalidArgument(
+          "unable to parse the specified CF option " + name);
+    }
+    compression_opts.enabled =
+        ParseBoolean("", value.substr(start, value.size() - start));
+  }
+  return Status::OK();
+}
+
```
Comment (siying) [inline options/options_helper.cc]: This function is moved to here from below.

File: HISTORY.md [hunk 2]
```diff
@@ -20,6 +20,7 @@
 * Basic support for user timestamp in iterator. Seek/SeekToFirst/Next and lower/upper bounds are supported. Reverse iteration is not supported. Merge is not considered.
 * When file lock failure when the lock is held by the current process, return acquiring time and thread ID in the error message.
 * Added a new option, best_efforts_recovery (default: false), to allow database to open in a db dir with missing table files. During best efforts recovery, missing table files are ignored, and database recovers to the most recent state without missing table file. Cross-column-family consistency is not guaranteed even if WAL is enabled.
+* make options.bottommost_compression, options.compression_opts and options.bottommost_compression_opts dynamically changeable.
```
Comment (riversand963) [inline HISTORY.md]: `s/make/Made/`.

File: options/options_helper.cc [hunk 3]
```diff
@@ -786,37 +786,117 @@ bool SerializeSingleOptionHelper(const char* opt_address,
...
-        ROCKS_LOG_WARN(info_log, "%s is a deprecated option and cannot be set",
-                       o.first.c_str());
-        continue;
-      }
-      bool is_ok = ParseOptionHelper(
-          reinterpret_cast<char*>(new_options) + opt_info.mutable_offset,
-          opt_info.type, o.second);
-      if (!is_ok) {
-        return Status::InvalidArgument("Error parsing " + o.first);
+      if (option_name == "bottommost_compression_opts") {
```
Comment (riversand963) [inline options/options_helper.cc]: How about defining a constant `kBottommostCompressionOptionsName`?

File: options/options_helper.cc [hunk 4]
```diff
@@ -786,37 +786,117 @@ bool SerializeSingleOptionHelper(const char* opt_address,
...
+        Status s =
+            ParseCompressionOptions(option_value, option_name,
+                                    new_options->bottommost_compression_opts);
+        if (!s.ok()) {
+          return s;
+        }
+      } else if (option_name == "compression_opts") {
```
Comment (riversand963) [inline options/options_helper.cc]: Ditto.

File: db/flush_job.cc [hunk 5]
```diff
@@ -384,7 +384,7 @@ Status FlushJob::WriteLevel0Table() {
           cfd_->GetName(), existing_snapshots_,
           earliest_write_conflict_snapshot_, snapshot_checker_,
           output_compression_, mutable_cf_options_.sample_for_compression,
-          cfd_->ioptions()->compression_opts,
+          mutable_cf_options_.compression_opts,
```
Comment (riversand963) [inline db/flush_job.cc]: `mutable_cf_options_` is a const reference, and `BuildTable()` does not hold the db mutex. What if a user thread calls `SetOptions` while flush job is accessing `mutable_cf_options_.compression_opts`?

Comment (riversand963) [reply db/flush_job.cc]: nvm, we do a copy before calling in `FlushMemTableToOutputFile` and `AtomicFlushMemTablesToOutputFiles`.

**Pull request closed without merging**